waspc/data/Generator/templates/react-app/src/actions/index.js

```diff
@@ -0,0 +1,74 @@
+import {
+  useMutation,
+  useQueryClient,
+} from 'react-query'
+
+export { configureQueryClient } from '../queryClient'
+
+export function useAction(actionFn, actionOptions) {
+  const queryClient = useQueryClient();
+
+  // TODO(filip): Do we allow all react query options or should we consider a list of allowed options?
+  let options = {}
```

Comment:
What are the pros/cons?
Also, from what I see, right now we allow only `.optimisticUpdates`, right?

Reply:
That's true, yes. I've experimented with merging objects, but we don't provide any configuration options at the moment.

Reply:
Pros/cons:
- Having more options
- Matching React Query's API 1 on 1 (easier to document, although we'd still have to cover the edge cases)
- Risking conflicts when users try to implement optimistic updates themselves

Versus (what if we don't support all react query options):
- Having a cleaner API (not introducing something we might not need or something users might use in ways we didn't expect and cause us problems)
- Having to put more effort into the docs (instead of just delegating it to React Query)
- More control

I'm leaning towards the second option

Reply:
I generally like leaning on external libraries for now, since we are yet early and have a lot of stuff on our plate to handle anyway. So if we can delegate and avoid writing and maintaining our docs, I vote for that. But if you feel that it is not worth it compared to control / safety we get, then let's go for the second one. You are the best judge here, I would just nudge you to imagine the practical use cases and pick the best option based on that.

Reply:
I ended up exposing our hook without the extra React Query stuff for now. They'll still be able to use React Query directly if they want it.